shared/templates/mount_option/oval.template

```diff
@@ -2,31 +2,66 @@
   <definition class="compliance" id="{{{ _RULE_ID }}}" version="2">
     {{{ oval_metadata(MOUNTPOINT ~ " should be mounted with mount option " ~ MOUNTOPTION ~ ".") }}}
     <criteria operator="OR">
-      <criterion comment="{{{ MOUNTOPTION }}} on {{{ MOUNTPOINT }}}"
-        test_ref="test_{{{ POINTID }}}_partition_{{{ MOUNTOPTIONID }}}_optional_{{{ MOUNT_HAS_TO_EXIST }}}"/>
+      <criteria operator="AND">
+         <!-- In case the mount point must exist in the system we will check both /proc/mounts and /etc/fstab
+              configuration for propers mount options to make sure the system is and will be configured correctly 
+              (after reboot) -->
```

Comment:
I miss the remeditation reason mentioned here. With this explanation it raises a question why not to remove the /proc/mounts check if the presistent configuration is only in /etc/fstab.

Reply:
Because the rule checks the online configuration as well and if it is not correct it will change / add the config and remount the mountpoint in question.

Reply:
> Because the rule checks the online configuration as well and if it is not correct it will change / add the config and remount the mountpoint in question.

Yes, and this is exactly what isn't obvious at all and therefore needs to be documented.

Reply:
Where exactly? The way the remediation(s) work has nothing to do with OVAL. And vice versa. Also, this is a template, which is used by many rules.

Reply:
> Where exactly?

shared/templates/mount_option/oval.template,  lines 6-8

> The way the remediation(s) work has nothing to do with OVAL.

You mentioned above that the reason why the OVAL is written this way is "It will trigger remediation that would do the right thing with the /etc/fstab." And also you said that the `/proc/mounts` is checked because if the online configuration isn't correct, the remediation will change / add the config and remount the mountpoint in question. Now, suddenly, the way the remediation(s) work has nothing to do with OVAL. This isn't consistent. Can you clarify this?

> Also, this is a template, which is used by many rules.

If it's used by many rules then it's important to make sure that people understand it correctly.

Reply:
So, why OVAL check and Bash remediation have no real connection between them? That's easy — there is no way to pass any information from OVAL check to the remediation. OVAL check is just a gateway switch, as it only controls if remediation will be executed or not. We can consider the OVAL check as a mere optimization because all Bash remediations should be written in a way that ensures proper fix for any combination of invalid configuration without any help from OVAL check.

For OVAL check, during the review, the only question that makes sense is: 'Does the check detect invalid configuration properly?'. How the configuration will be remediated (if ever) is not relevant for the check.

For a remediation the question is: 'Does the remediation bring the system configuration into the proper state?'. We can't assume that system is or is not in a proper state. And we must do it without any information from checking system.

Reply:
So, getting back to the original question. Is a mounted mount point with incorrect options a valid system state as it is perceived by the rules that use this template or not? Please note, that this on-line /proc/mount check was in the template from the beginning, and I would really like to have a good reason to remove the behavior that many rules rely on.

Reply:
This is now a bit irrelevant. Resolving.